lib/fetch/dataURL.js

```diff
@@ -125,11 +125,12 @@ function URLSerializer (url, excludeFragment = false) {
     return href
   }
 
-  const hash = href.lastIndexOf('#')
-  if (hash === -1) {
-    return href
-  }
-  return href.slice(0, hash)
+  // Note: Since there are at least two characters in the url,
+  // the scheme and the U+003A (:) character,
+  // the hash searches after these.
+  const hash = href.indexOf('#', 2)
+
+  return hash === -1 ? href : href.substring(0, hash)
```

Comment:
Isn't something like this more reliable?

```
if (!url.hash) {
  return href
}

return href.substring(0, -href.hash.length)
```

Reply:
I used `String#indexOf` because I felt that you were using `String#lastIndexOf` intentionally.
I was looking at the commit log and it doesn't seem intentional, so I will do so.